hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/execution/init/Contexts.java

```diff
@@ -185,13 +188,17 @@ private File extractFileToDisk(String id) {
             IMap<String, byte[]> map = jetInstance().getMap(jobResourcesMapName(jobId()));
             try (IMapInputStream inputStream = new IMapInputStream(map, fileKeyName(id))) {
                 String prefix = "jet-" + jetInstance().getName() + "-" + idToString(jobId()) + "-" + id;
-                Path directory = Files.createTempDirectory(prefix);
+                Path directory = Files.createTempDirectory(sanitizeFilename(prefix));
                 unzip(inputStream, directory);
                 return directory.toFile();
             } catch (IOException e) {
                 throw ExceptionUtil.rethrow(e);
             }
         }
+
+        private static String sanitizeFilename(String proposed) {
+            return proposed.replace('/', '_');
```

Comment:
This is a good idea. The `id` is user-supplied, it can be whatever. We should whitelist allowed characters and replace all others. Also maybe trim it to up to 20 chars or so.